jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunExecutorActor.scala

```diff
@@ -30,17 +32,22 @@ class JobRunExecutorActor(
     driverHolder:               MarathonSchedulerDriverHolder,
     clock:                      Clock,
     val behavior:               Behavior
-) extends Actor with Stash with ActorLogging with ActorBehavior {
+)(implicit scheduler: Scheduler) extends Actor with Stash with ActorLogging with ActorBehavior {
   import JobRunExecutorActor._
   import JobRunPersistenceActor._
   import TaskStates._
+  import context.dispatcher
+
+  private[impl] var startingDeadlineTimer: Option[Cancellable] = None
```

Comment:
What do you think about moving this var into the receive function parameter? less mutable state, less problems :)

Or we can use heavy guns like akka-FSM :)

Reply:
Recommendation to avoid Akka FSM. It doesn't end well.